File: EncontreMaiorSubstring.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./EncontreMaiorSubstring.js')

TEST('Starting EncontreMaiorSubstring test...', (t) => {
    //Test 1
    t.assert(OBJ.findTheBiggestSubstring("abcdef", "cdofhij") === 2, "Expect: 2")
    //Test 2
    t.assert(OBJ.findTheBiggestSubstring("TWO", "FOUR") === 1, "Expect: 1")
    //Test 3
    t.assert(OBJ.findTheBiggestSubstring("abracadabra", "open") === 0, "Expect: 0")
    //Test 4
    t.assert(OBJ.findTheBiggestSubstring("Hey This java is hot", "Java is a new paradigm") === 7, "Expect: 7")

    t.end()
})

File: GO/EncontreMaiorSubstring_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestEncontreMaiorSubstring(t *testing.T) {
	tables := []struct {
		w1   string
		w2   string
		want int
	}{
		{"abcdef", "cdofhij", 2},
		{"TWO", "FOUR", 1},
		{"abracadabra", "open", 0},
		{"Hey This java is hot", "Java is a new paradigm", 7},
	}

	for _, table := range tables {
		got := findTheBiggestSubstring(table.w1, table.w2)

		if got != table.want {
			t.Errorf("Waiting for this %.d but the return was this: %.d", table.want, got)
		}
	}
}
